client/tinychain/math/operator.py

```diff
@@ -106,7 +106,7 @@ def backward(self, variable):
         return (subject @ self.args) + (self.subject @ arg)
```

Comment:
`backward` is useful to calculate the (partial) derivative of an `Operator`, e.g. to calculate the derivative of a cost function automatically, or in a situation where a mathematical model needs to calculate the derivative of its input. I can rename it to `derivative` if you think that's more descriptive. One way I think it might be useful is to define a general solution to calculating the gradients of an `Operator`, which might solve your problem with `Add`. Consider:

```python
def gradients(operator, d_loss):
    if isinstance(operator.subject, Variable):
        return d_loss * operator.backward(operator.subject)
    # ...
```